### ISSUE TITLE
Run Models in Parallel

### ISSUE THREAD

**Issue by scybersec** [awr7/MoM#2]
Running models sequentially takes more time as we need to wait for the output of one before starting the next model API call. This process is O(n) because for n models, we run through each one sequentially. However, by executing them in parallel, we can improve the efficiency to a constant O(1).

---

$$
\large \text {{Idea}}
$$

To run models in parallel, we can utilize `ThreadPoolExecutor` to manage threads, allow us to send API requests to all the models simultaneously. 

To implement parallel execution in our code, we can use **`ThreadPoolExecutor`** from the **`concurrent.futures`** module. First, we'll initiate a **`ThreadPoolExecutor`** to create a pool of threads. Then, we'll submit each model's API call as a separate task to the executor. By using **`executor.submit`**, we can run these tasks concurrently, allowing all API requests to be sent simultaneously. We can gather the results as they complete and handle any exceptions that might occur during execution. This approach will significantly improve the efficiency of our system.

$$
\large \text {{Initiate Parallel Execution}}
$$

```mermaid
graph TD
    A(Initiate ThreadPoolExecutor) --> B(Submit Model Tasks)
    B --> C(Execute Tasks Concurrently)
    C --> D(Gather Results as Completed)
    D --> E[Handle Results and Exceptions]

```

$$
\large \text {{ThreadPoolExecutor Docs}}
$$

https://docs.python.org/3/library/concurrent.futures.html
